### DISCUSSION TITLE
use loaderData in Links function, similar to Meta function

### DISCUSSION THREAD

**Discussion by mechanicMo** [remix-run/remix#4366] [Proposals]
i noticed that the meta function now receives loaderData from the route - amazing change. is this also possible to implement for the links function? for example, i would like to be able to dynamically set a favicon url from loaderData

### Working Meta function:
```
export const meta: MetaFunction = ({
    data,
}) => ({
    title: data.myConfig.myTitle,
});
```

### Requested Links function:
```
export const links: LinksFunction = ({ data }) => [ 
    {
        rel: 'icon',
        href: data.myFaviconURL,
    },
];
```

**Comment by sergiodxa** [comment]
The `MetaFunction` has always received the loader data from the route, it's not a new change.

The `LinksFunction` can't receive it because that would make it impossible to prefetch links when using `<Link prefetch>`, if you need to data to know the links to prefetch Remix can't use simple `<link rel="prefetch">` tags and it will have to send a `fetch` request to the loader, get the data, call links and then know what to prefetch.

You can use [Remix Utils's DynamicLinks](https://github.com/sergiodxa/remix-utils#dynamiclinks) to create links based on your route data.

**Reply by mechanicMo** [reply]
Thanks @sergiodxa. this is util solves my use case!

**Comment by JulienMeziere** [comment]
Seems like the DynamicLinks solution doesn't exist anymore
I solved the issue following the solution in that thread: https://github.com/remix-run/remix/discussions/6977
